algorithm.go

```diff
@@ -72,6 +98,41 @@ func (a Algorithm) String() string {
 	}
 }
 
+// MarshalCBOR marshals the Algorithm as a CBOR int.
+func (a Algorithm) MarshalCBOR() ([]byte, error) {
+	return encMode.Marshal(int(a))
+}
+
+// UnmarshalCBOR populates the Algorithm form the provided CBOR value (must be
+// int or tstr).
+func (a *Algorithm) UnmarshalCBOR(data []byte) error {
+	var raw IntOrStr
+
+	if err := raw.UnmarshalCBOR(data); err != nil {
+		return fmt.Errorf("invalid algorithm value: %w", err)
+	}
+
+	if raw.IsString() {
+		v, err := AlgorithmFromString(raw.String())
+
+		if err != nil {
+			return err
+		}
+
+		*a = v
+	} else {
+		v := raw.Int()
+
+		if (v < -39 || v > -35) && (v < -8 || v > -7) {
```

Comment:
this could be more tight with the `Algorithm` type and constants definitions above (e.g., adding a `IsSupported()` method)

Reply:
Const names would be useful here, as we're defining nurical ranges here, not refering to specific values with meanings attached.  `v < AlgorithmPS512 || v > AlgorithmES384`  makes less sense than `v < -39 || v > -35`

Reply:
Not really -- because we're not comparing the names directly -- there is no inherent ordering to algorithm values, so it's not clear why `v` needs to less `AlgorithmPS512` but creater than `AlgorithmES384`, and in order for the check to make any sense,  we need to assume that `AgorithmPS512 > AlgorithmES384`, and there is no reason why that has to be the case. 

Conversely, leaving the literal values in makes it clear tha we're trying to make sure that v is in the range `-39 <= v <= -35`.

Names would noly make sense in such a comparison if there was an inherent boundary implied by the name (e.g. if it were a sentinel -- i.e. `v > AlgorithmInvalid` makse sense), but that is not the case here.